vortex-filtering/include/vortex_filtering/probability/poisson.hpp

```diff
@@ -0,0 +1,60 @@
+/**
+ * @file poisson.hpp
+ * @author Eirik Kolås
+ * @brief A class for representing a Poisson distribution. Used for modeling clutter
+ * @version 0.1
+ * @date 2023-10-26
+ *
+ * @copyright Copyright (c) 2023
+ *
+ */
+#pragma once
+#include <cmath>
+
+namespace vortex {
+namespace prob {
+
+class Poisson {
+public:
+  Poisson(double lambda) : lambda_(lambda) {}
+
+  /** Calculate the probability of x given the Poisson distribution
+   * @param x
+   * @return double
+   */
+  double pr(int x) const { return std::pow(lambda_, x) * std::exp(-lambda_) / factorial(x); }
+
+  /** Calculate the mean of the Poisson distribution
+   * @return double mean
+   */
+  double mean() const { return lambda_; }
+
+  /** Calculate the variance of the Poisson distribution
+   * @return double variance
+   */
+  double cov() const { return lambda_; }
+
+  /** Parameter lambda of the Poisson distribution
+   * @return double lambda
+   */
+  double lambda() const { return lambda_; }
+
+private:
+  const double lambda_;
+
+  /** Calculate the factorial of x
+   * @param x
+   * @return double factorial
+   */
+  double factorial(int x) const
+  {
+    double factorial = 1;
+    for (int i = 1; i <= x; i++) {
+      factorial *= i;
+    }
+    return factorial;
+  }
```

Comment:
Using this as an example, but there are a lot of implementations-from-scratch here, is there a reason behind that vs using boost:: or any other well-established libraries for it?

Reply:
Not really. Haven't been looking at other libraries. Have also avoided using boost

Reply:
Fair, as long as its performant its cool to have most of it built from scratch 😎